# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/TransportSecurityInfo.cpp
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e security/manager/ssl/TransportSecurityInfo.cpp
--- a/security/manager/ssl/TransportSecurityInfo.cpp	Mon Oct 21 19:33:02 2019 +0000
+++ b/security/manager/ssl/TransportSecurityInfo.cpp	Mon Oct 21 19:49:01 2019 +0000
@@ -891,10 +891,14 @@
 }
 
 NS_IMETHODIMP
-TransportSecurityInfo::GetFailedCertChain(nsIX509CertList** _result) {
-  MOZ_ASSERT(_result);
-  return TransportSecurityInfo::ConvertCertArrayToCertList(mFailedCertChain,
-                                                           _result);
+TransportSecurityInfo::GetFailedCertChain(
+    nsTArray<RefPtr<nsIX509Cert>>& aFailedCertChain) {
+  MOZ_ASSERT(aFailedCertChain.IsEmpty());
+  if (!aFailedCertChain.IsEmpty()) {
+    return NS_ERROR_INVALID_ARG;
+  }
+  aFailedCertChain.AppendElements(mFailedCertChain);
+  return NS_OK;
 }
 
 nsresult TransportSecurityInfo::SetFailedCertChain(
@@ -927,11 +931,14 @@
 }
 
 NS_IMETHODIMP
-TransportSecurityInfo::GetSucceededCertChain(nsIX509CertList** _result) {
-  NS_ENSURE_ARG_POINTER(_result);
-
-  return TransportSecurityInfo::ConvertCertArrayToCertList(mSucceededCertChain,
-                                                           _result);
+TransportSecurityInfo::GetSucceededCertChain(
+    nsTArray<RefPtr<nsIX509Cert>>& aSucceededCertChain) {
+  MOZ_ASSERT(aSucceededCertChain.IsEmpty());
+  if (!aSucceededCertChain.IsEmpty()) {
+    return NS_ERROR_INVALID_ARG;
+  }
+  aSucceededCertChain.AppendElements(mSucceededCertChain);
+  return NS_OK;
 }
 
 nsresult TransportSecurityInfo::SetSucceededCertChain(